tx/textual/valuerenderer/coin_test.go

```diff
@@ -69,6 +71,14 @@ func TestCoinJsonTestcases(t *testing.T) {
 				ctx := context.WithValue(context.Background(), mockCoinMetadataKey(tc.Proto.Denom), tc.Metadata)
 				screens, err := vr.Format(ctx, protoreflect.ValueOf(tc.Proto.ProtoReflect()))
 
+				var baseDenom string
+				if tc.Metadata != nil {
+					baseDenom = tc.Metadata.Display
+				} else {
+					baseDenom = strings.Split(screens[0].Text, " ")[1]
+				}
+				ctx = context.WithValue(context.Background(), mockCoinMetadataKey(baseDenom), tc.Metadata)
```

Comment:
Proposing shorter code for the same result:

```suggestion
if tc.Metadata != nil {
  ctx = context.WithValue(ctx, mockCoinMetadataKey(tc.Metadata.Display), tc.Metadata)
}
```

Reply:
Also, I think moving it up 2 lines would make more sense, right after `ctx :=`, so that we set up context in one place

Reply:
when metadata is nil querying metadata will return nil(since we didn't set any) in parse method, hence it will break few scenarios.
to handle that I added this in else case
```
baseDenom = strings.Split(screens[0].Text, " ")[1]
```

iit's breaking few tests if I remove these lines.

Reply:
See my proposal here: https://github.com/cosmos/cosmos-sdk/compare/ap/parse-coins...am/parse-coins-2?expand=1. I don't think it's breaking tests